docs/FAQ/FAQ-Wallet.md

```diff
@@ -1,133 +1,80 @@
-
-For that reason, it's highly recommended that you use [Electrum wallet](https://electrum.org/). To set increase the gap limit in Electrum, [follow this video](https://www.youtube.com/watch?v=Fi3pYpzGmmo)
+If you're using an [existing software or a hardware wallet](/WalletSetup/#use-an-existing-wallet) with your BTCPay Server, you may experience a discrepancy between balance in your BTCPay wallet and the external wallet's web, destkop or mobile app. This discrepancy is usually related to a **gap-limit** issue. 
 
-Enter following commands in Electrum console:
+### The gap limit problem
 
-```
- wallet.change_gap_limit(100)
- wallet.storage.write()
-```
-Restart your Electrum and verify that the newly set gap limit is correct by entering:
+The majority of third party wallets are [SPV wallets](https://en.bitcoinwiki.org/wiki/Simplified_Payment_Verification), which share a node between many users. To prevent performance issues, SPV wallets limit the amount (typically 20) of addresses without balance that they track on the blockchain. BTCPay Server generates a new address for every invoice. 
 
-`wallet.gap_limit`
+With above in mind, after BTCPay Server generates 20 consecutive unpaid invoices, the external wallet stops fetching the transactions, assuming no new transactions occurred. Once 21st, 22nd, etc invoices are paid, your external wallet won't show them.
 
-There's no good answer to how much you should set the gap limit to. Most merchants set 100-200. If you're a big merchants with high transaction volume, you can try with an even higher gap limit.
+On the other hand, internally, BTCPay Server wallet tracks any address it generates itself along with a much greater gap limit. It does not rely on a third-party and is able to always show a correct balance.
 
-Be aware that :
+### The gap limit solution
 
-* A higher gap limit may slow down the performance of your wallet
-* Not all wallets support the increased gap limit. If you import an Electrum recovery seed into another wallet, you may not see all the funds again.
+It's not easy to solve the gap limit problem. You have two options:
 
-When an invoice is created in BTCPay, it does it for all coins you have setup. You may want to increase the gap limit for altcoins as well in their supported wallets.
+1. Increase the gap limit in your existing (external) wallet
+2. Use internal BTCPay Server wallet
 
-If you do not see your funds yet, you may have set up your derivation scheme incorrectly.
+#### 1. Increasing the gap limit
 
-## Why is my Ledger not detected by BTCPay Server?
+If your [external/existing wallet](/WalletSetup.md/#use-an-existing-wallet) allows gap-limit configuration, the easy fix is to increase it. However, majority of wallets do not allow this. 
```

Comment:
Is there a way to see the current largest gap from the btcpay gui? It might be useful to show this somewhere, so users can know what gap limit to configure their external wallets with.

Reply:
We do have a wallet rescan (utxoscan) feature, is that what you mean?
![rescan](https://user-images.githubusercontent.com/36959754/84591443-ae146900-ae3e-11ea-9e73-3953449ed59e.PNG)

Reply:
I meant a way to see the current largest gap that exists in the wallet, similarly to [this](https://github.com/shesek/bwt#get-hdfingerprintgap).

Reply:
I don't think we do, interesting. Ping @Kukks ^

Reply:
No we don't particularly keep track of this